pkg/reconciler/revision/resources/deploy.go

```diff
@@ -170,6 +172,68 @@ func makePodSpec(rev *v1alpha1.Revision, loggingConfig *logging.Config, tracingC
 	return podSpec, nil
 }
 
+func verifySecrets(rev *v1alpha1.Revision, secretLister v1.SecretLister) *apis.FieldError {
+	var apiErrs *apis.FieldError
+	// Check secrets in volume source
+	for i, volume := range rev.Spec.Volumes {
+		if secret := volume.VolumeSource.Secret; secret != nil && (secret.Optional == nil || *secret.Optional == false) {
+			if _, err := secretLister.Secrets(rev.Namespace).Get(secret.SecretName); err != nil {
+				newErr := (&apis.FieldError{
+					Message: err.Error(),
+					Paths:   []string{apis.CurrentField},
+				}).ViaField("secretName").ViaField("volumeSource").
+					ViaFieldIndex("volumes", i).ViaField("spec")
+				if apiErrs == nil {
+					apiErrs = newErr
+				} else {
+					apiErrs = apiErrs.Also(newErr)
+				}
+			}
+		}
+		if projected := volume.VolumeSource.Projected; projected != nil {
+			for j, sources := range projected.Sources {
+				if secret := sources.Secret; secret != nil && (secret.Optional == nil || *secret.Optional == false) {
```

Comment:
This looks like 3 times exactly the same code, could it be simplified to something like:

```golang
for i, volume := range rev.Spec.Volumes {
  if secret := volume.VolumeSource.Secret; secret != nil {
    apiErrs.Also(verifySecret(secret).ViaField("secretName").ViaField("volumeSource").ViaFieldIndex("volumes", i)
  }
}

...
```